src/tensor_computes/LBMFixedZerothOrderBC.C

```
@@ -0,0 +1,388 @@
+/**********************************************************************/
+/*                    DO NOT MODIFY THIS HEADER                       */
+/*             Swift, a Fourier spectral solver for MOOSE             */
+/*                                                                    */
+/*            Copyright 2024 Battelle Energy Alliance, LLC            */
+/*                        ALL RIGHTS RESERVED                         */
+/**********************************************************************/
+
+#include "LBMFixedZerothOrderBC.h"
+#include "LatticeBoltzmannProblem.h"
+#include "LatticeBoltzmannStencilBase.h"
+
+using namespace torch::indexing;
+
+registerMooseObject("SwiftApp", LBMFixedZerothOrderBC9Q);
+registerMooseObject("SwiftApp", LBMFixedZerothOrderBC19Q);
+registerMooseObject("SwiftApp", LBMFixedZerothOrderBC27Q);
+
+template <int dimension>
+InputParameters
+LBMFixedZerothOrderBCTempl<dimension>::validParams()
+{
+  InputParameters params = LBMBoundaryCondition::validParams();
+  params.addClassDescription("LBMFixedZerothOrderBC object");
+  params.addRequiredParam<TensorInputBufferName>("f", "Input buffer distribution function");
+  params.addRequiredParam<std::string>("value", "Fixed input value");
+  return params;
+}
+
+template <int dimension>
+LBMFixedZerothOrderBCTempl<dimension>::LBMFixedZerothOrderBCTempl(
+    const InputParameters & parameters)
+  : LBMBoundaryCondition(parameters),
+    _f(getInputBufferByName(getParam<TensorInputBufferName>("f"))),
+    _grid_size(_lb_problem.getGridSize()),
+    _value(_lb_problem.getConstant<Real>(getParam<std::string>("value")))
+{
+}
+
+template <>
+void
+LBMFixedZerothOrderBCTempl<9>::frontBoundary()
+{
+  // There is no front boundary in 2D
+  mooseError("There is no front boundary in 2 dimensions.");
+}
+
+template <>
+void
+LBMFixedZerothOrderBCTempl<19>::frontBoundary()
+{
+  // TBD
```

same here. error out rather than silently not implement